src/test/java/Tests/Task2_Tests.java

```diff
@@ -1,93 +1,51 @@
 package Tests;
 
+import Pages.HomePage;
+import Pages.Task2_Page;
 import config.TestConfig;
-import org.junit.After;
 import org.junit.Assert;
-import org.junit.Before;
 import org.junit.Test;
-import org.openqa.selenium.By;
 import org.openqa.selenium.WebDriver;
 import org.openqa.selenium.WebElement;
-import org.openqa.selenium.chrome.ChromeDriver;
-
-import java.util.concurrent.TimeUnit;
 
 public class Task2_Tests extends TestConfig {
 
     // MARK: - Properties
 
     WebDriver driver;
 
-    // MARK: - Lifecycle
-
-    @Before
-    public void setUp() {
-        System.setProperty("Webdriver ChromeDriver", "C:\\Users\\Aga\\Desktop\\FirstTestWeb\\src\\test\\java\\chromedriver.exe");
-        driver = new ChromeDriver();
-        driver.manage().window().maximize();
-        driver.manage().timeouts().implicitlyWait(10000, TimeUnit.MILLISECONDS);
-        driver.get("http://the-internet.herokuapp.com/");
-    }
-
-    @After
-    public void tearDown() {
-        driver.quit();
-    }
+    private HomePage homePage;
+    private Task2_Page taskTwoPage;
 
     // MARK: - Test cases
 
     @Test
     public void checkboxesTest() throws InterruptedException {
-        this.tapCheckboxTask();
-        this.isCheckboxTaskHeaderDisplayed();
-
-        WebElement checkbox_first = driver.findElement(By.xpath("//form[@id='checkboxes']/input[@type='checkbox'][1]"));
-        WebElement checkbox_second = driver.findElement(By.xpath("//form[@id='checkboxes']/input[@type='checkbox'][2]"));
+        homePage.openSecondTask();
+        taskTwoPage.isCheckboxHeaderDisplayed();
+        Assert.assertTrue(taskTwoPage.isChecboxFirstDisplayed());
+        Assert.assertTrue(taskTwoPage.isChecboxSecondDisplayed());
 
-        checkbox_first.isDisplayed();
-        checkbox_second.isDisplayed();
-
-        Assert.assertFalse(this.isCheckboxSelected(checkbox_first));
-        Assert.assertTrue(this.isCheckboxSelected(checkbox_second));
+        Assert.assertFalse(taskTwoPage.isCheckboxSelected(taskTwoPage.getCheckbox_first()));
```

Comment:
bardzo dziwna konstrukcja - powinno coś takiego być ukryte niżej, na poziomie Page'a, nie tutaj w teście.

Reply:
dotyczy też linii 30.

Reply:
Jeśli chciałabym umieścić to na poziomie Page'a to wydaje mi się że bym zdublowała kod. 
Oczywiście można by się nad tym zastanowić jeśli miałoby powstać więcej testów dot. checkboxów.
Wydaje mi się, że metoda .isCheckboxSelectes(WebElement element) załatwia sprawę bo jest reużywalna, ponieważ przekazuję do niej argument, który jako jedyny by zmieniał się w tej metodzie. W przeciwnym razie miałabym dwie takie same metody dla różnych elementów - co powoduje duplikowanie kodu.